tests/ui/drop/issue-23338-ensure-param-drop-order.rs

```diff
@@ -144,7 +144,9 @@ pub mod d {
                 let trail = first_avail();
                 let ctr = counter;
                 counter += 1;
+                //~^ WARN creating a mutable reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.